### PR TITLE
Fix support for static provisioning

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

See https://github.com/crossplaneio/crossplane-runtime/pull/24 for context. That PR must first be merged, and then this PR updated to depend on the latest build, before this PR can be merged.

Reviewers, please double check the kinds I've provided to each predicate to ensure they make sense; it's especially easy to mix up `FooGroupVersionKind` with `FooClassGroupVersionKind`.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: pkg/controller/azure/cache/claim.go [hunk 1]
```diff
@@ -39,22 +39,36 @@ type RedisClaimController struct{}
...
+	name := strings.ToLower(fmt.Sprintf("%s.%s.%s",
+		cachev1alpha1.RedisClusterKind,
+		v1alpha2.RedisKind,
+		v1alpha2.Group))
+
 	r := resource.NewClaimReconciler(mgr,
 		resource.ClaimKind(cachev1alpha1.RedisClusterGroupVersionKind),
-		resource.ClassKinds{Portable: cachev1alpha1.RedisClusterGroupVersionKind, NonPortable: v1alpha2.RedisClassGroupVersionKind},
+		resource.ClassKinds{
+			Portable:    cachev1alpha1.RedisClusterGroupVersionKind,
```
Comment (hasheddan) [inline pkg/controller/azure/cache/claim.go]: Finally found one :) should be `cachev1alpha1.RedisClusterClassGroupVersionKind`

File: pkg/controller/azure/cache/claim.go [hunk 2]
```diff
@@ -39,22 +39,36 @@ type RedisClaimController struct{}
...
 			resource.ManagedConfiguratorFn(ConfigureRedis),
 			resource.NewObjectMetaConfigurator(mgr.GetScheme()),
 		))
 
-	name := strings.ToLower(fmt.Sprintf("%s.%s", cachev1alpha1.RedisClusterKind, controllerName))
+	p := resource.NewPredicates(resource.AnyOf(
+		resource.HasManagedResourceReferenceKind(resource.ManagedKind(v1alpha2.RedisGroupVersionKind)),
+		resource.HasDirectClassReferenceKind(resource.NonPortableClassKind(v1alpha2.RedisClassGroupVersionKind)),
+		resource.HasIndirectClassReferenceKind(mgr.GetClient(), mgr.GetScheme(), resource.ClassKinds{
+			Portable:    cachev1alpha1.RedisClusterGroupVersionKind,
```
Comment (hasheddan) [inline pkg/controller/azure/cache/claim.go]: `cachev1alpha1.RedisClusterClassGroupVersionKind` here as well

File: pkg/controller/azure/database/claim.go [hunk 3]
```diff
@@ -105,22 +119,36 @@ type MySQLInstanceClaimController struct{}
...
+	name := strings.ToLower(fmt.Sprintf("%s.%s.%s",
+		databasev1alpha1.MySQLInstanceKind,
+		v1alpha2.MysqlServerKind,
+		v1alpha2.Group))
+
 	r := resource.NewClaimReconciler(mgr,
 		resource.ClaimKind(databasev1alpha1.MySQLInstanceGroupVersionKind),
-		resource.ClassKinds{Portable: databasev1alpha1.MySQLInstanceGroupVersionKind, NonPortable: v1alpha2.SQLServerClassGroupVersionKind},
+		resource.ClassKinds{
+			Portable:    databasev1alpha1.MySQLInstanceGroupVersionKind,
```
Comment (hasheddan) [inline pkg/controller/azure/database/claim.go]: Should be `databasev1alpha1.MySQLInstanceClassGroupVersionKind`

Comment (hasheddan) [reply pkg/controller/azure/database/claim.go]: This one is a good example of proposal in [comment](https://github.com/crossplaneio/stack-gcp/pull/24#discussion_r324423610) due to the fact that the correct `Portable` class kind is used in the predicates below. Still would be possible to use wrong kind, but would at least reduce having to make sure it is right twice :)

**Comment by negz** [comment]
Thanks for the eagle eyed review @hasheddan! PTAL.

**Pull request merged**